### PR TITLE
docs: update kernel minimum requirement

### PR DESCRIPTION
**Pull request by dreamerlzl**
Related: #1769

According to [this](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md), 5.4 only introduces `BPF_MAP_TYPE_DEVMAP_HASH` for eBPF. Any explanation for why requiring 5.4 instead of the original 4.18+(e.g. Almalinux 8.7 would be excluded)?

### CODE REVIEW

**Comment by javierhonduco** [comment]
The introduction of `BPF_MAP_TYPE_DEVMAP_HASH` is unrelated to our requirements. We mainly have two constraints here:
- We require BTF support in the kernel. This was added in `4.18`, but wasn't enabled by most distros until later. We could download the needed BTF files on demand, and in fact we started this feature in https://github.com/parca-dev/parca-agent/pull/1131, but we have higher priority items and it's not likely going to be worked on in the near future;
- We need reasonably modern kernels that are ok with some BPF features we use, such as bounded loops, introduced in `5.3`;

Theoretically, the Agent should run in 5.3 or greater, but we only guarantee that the agent will work in the kernels we test on. Any kernel version in between the supported ones should work, minus bugs in the kernel.

We've done some experiments to allow the Agent to work in `4.x` (we require tail-calls, released in 4.2, and there were some fixes around 4.10 that we need, too) but this is not a priority for us and it can introduce a performance regression for all kernels (1M instructions were only added in `5.2`) , including modern ones that have features that allow our programs to run more efficiently.

All these versions have been pulled from https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Parca-agent follows [CNCF Code of Conduct](https://github.com/cncf/foundation/bl
 <!-- TODO: Add code of conduct info-->
 # Prerequisites
 
-- Linux Kernel version 4.18+
+- Linux Kernel version 5.4+
```
Comment (javierhonduco) [inline CONTRIBUTING.md:17]:
```suggestion
- Linux Kernel version 5.3+ with BTF
```

File: README.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ The collected data can be viewed locally via HTTP endpoints and then be configur
 
 ## Requirements
 
-* Linux Kernel version 4.18+
+* Linux Kernel version 5.4+
```
Comment (javierhonduco) [inline README.md:14]: Same as above

File: docs/design.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ Parca Agent implements a sampling profiler, to sample stack traces [19 times per
...
+Parca Agent uses BPF CO-RE (Compile Once – Run Everywhere) using [libbpf](https://github.com/libbpf/libbpf), pre-compiles all BPF programs, and statically embeds them in the target binary, from where it is loaded via libbpf when used. This means that Parca Agent does not need to compile the BPF program at startup or runtime like when using [bcc-tools](https://github.com/iovisor/bcc/tree/master/tools), meaning no Clang & LLVM, nor kernel headers need to be installed on the host. The only requirement is a [BTF](https://www.kernel.org/doc/html/latest/bpf/btf.html)-capable Kernel (Linux Kernel 5.4+).
```
Comment (javierhonduco) [inline docs/design.md:7]: Same

File: docs/security.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ Parca Agent binaries and container image build processes have been specifically
...
+Parca Agent uses BPF CO-RE (Compile Once – Run Everywhere) using [libbpf](https://github.com/libbpf/libbpf), and pre-compiles all BPF programs, and statically embeds them in the target binary, from where it is loaded via libbpf when used. This means that Parca Agent does not need to compile the BPF program at startup or runtime like when using [bcc-tools](https://github.com/iovisor/bcc/tree/master/tools), meaning no Clang & LLVM, nor kernel headers need to be installed on the host. The only requirement is a [BTF](https://www.kernel.org/doc/html/latest/bpf/btf.html) capable Kernel (Linux Kernel 5.4+).
```
Comment (javierhonduco) [inline docs/security.md:16]: Same as above

**Comment by dreamerlzl** [comment]
Thank you for the explanation. Updated.

**Review by javierhonduco — APPROVED** [review]
LGTM, thanks for the change!

**Pull request merged**